### ISSUE TITLE
🐛 Bug: "This package is based on" notice replaces create-typescript-app with --repository

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/create-typescript-app#750]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Running any setup script on a repository should put the following notice on the bottom of the README.md:

```plaintext
<!-- You can remove this notice if you don't want it 🙂 no worries! -->

> 💙 This package is based on [@JoshuaKGoldberg](https://github.com/JoshuaKGoldberg)'s [template-typescript-node-package](https://github.com/JoshuaKGoldberg/template-typescript-node-package).
```

### Actual

```md
<!-- You can remove this notice if you don't want it 🙂 no worries! -->

> 💙 This package is based on [@JoshuaKGoldberg](https://github.com/JoshuaKGoldberg)'s [all-contributors-auto-action](https://github.com/JoshuaKGoldberg/all-contributors-auto-action).
```

Lol. The find-and-replace script that switches from the template title to the provided repository title (`--repository`) must be catching this.

### Additional Info

Discovered on https://github.com/JoshuaKGoldberg/all-contributors-auto-action/tree/1ce7a4d2df5fd69765b7d407654847fde9e30340:

```shell
git clone https://github.com/JoshuaKGoldberg/all-contributors-auto-action
cd all-contributors-auto-action
git checkout 1ce7a4d2df5fd69765b7d407654847fde9e30340
pnpm i
npx create-typescript-app --base everything --mode migrate
```
